Jenkinsfile

```diff
@@ -1253,6 +1254,71 @@ pipeline {
                         }
                     }
                 }
+                stage('win-2022-c8d') {
+                    when {
+                        beforeAgent true
+                        expression { params.windows2022containerd }
+                    }
+                    environment {
+                        DOCKER_BUILDKIT        = '0'
+                        DOCKER_DUT_DEBUG       = '1'
+                        SKIP_VALIDATION_TESTS  = '1'
+                        SOURCES_DRIVE          = 'd'
+                        SOURCES_SUBDIR         = 'gopath'
+                        TESTRUN_DRIVE          = 'd'
+                        TESTRUN_SUBDIR         = "CI"
+                        // TODO switch to mcr.microsoft.com/windows/servercore:2022 once published
+                        WINDOWS_BASE_IMAGE     = 'mcr.microsoft.com/windows/servercore/insider'
+                        // Available tags can be found at https://mcr.microsoft.com/v2/windows/servercore/insider/tags/list
+                        WINDOWS_BASE_IMAGE_TAG = '10.0.20344.1'
```

Comment:
Until server template is updated this one need point to `10.0.20295.1`

Reply:
@StefanScherer is there 20344 build version of server templates available already on Azure?

Reply:
@olljanat I'll check if there are newer templates.
I checked a few weeks ago, but that version broke our packer build to prepare the custom VM image. 
But I can try again and see if it works again.

Reply:
@StefanScherer any news?

Reply:
@olljanat Yes, I've created a new Azure VM image with Windows Server 2022 preview, Version `10.0.20348.51` and Docker 20.10.7 preinstalled. I cannot specify any other OS version.
Would this new VM image help you?

Reply:
I've checked the image tags, but the best I can see still is `10.0.20344.1`, so it would run in Hyper-V isolation, right?

Reply:
Oh, that's odd (and unfortunate) that they don't have images for that but that there are machines with that newer version 😞 

We currently don't run CI with Hyper-V on this repo; see https://github.com/moby/moby/pull/42277.

However, given that `10.0.20348.51` > `10.0.20344.1`, I _think_ that would work with process isolation; newer kernel supports older images (I thinK?). Let me try updating

Reply:
Unfortunately it doesn't work with process isolation:
```
d:\jenkins\workspace\test-azure-win-2022-agents>docker run mcr.microsoft.com/windows/servercore/insider:10.0.20344.1 
Unable to find image 'mcr.microsoft.com/windows/servercore/insider:10.0.20344.1' locally
10.0.20344.1: Pulling from windows/servercore/insider
a2ba77189187: Pulling fs layer
a2ba77189187: Verifying Checksum
a2ba77189187: Download complete
a2ba77189187: Pull complete
Digest: sha256:56bac714fc595e91860d6c9c30a6ba4b1da2eb1278ac95501bc413c484a46ad5
Status: Downloaded newer image for mcr.microsoft.com/windows/servercore/insider:10.0.20344.1
docker: Error response from daemon: hcsshim::CreateComputeSystem 4f2d7ff8b86dc6471c2680dc9da820522bc54f22255f415a8af685991468f875: The container operating system does not match the host operating system.
script returned exit code 125
```

Reply:
Hm. Official Windows Server Insider Preview [download page](https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewserver) offers 20344 build but they have later version on Azure? 😮

Reply:
Yes, I checked again a vanilla machine, but it's already 10.0.2348.51. 😿

Reply:
@thaJeztah so can you update this one to use `10.0.20295.1` now and update both of those to latest on #42527 when new version is available?

Reply:
Process isolation build IDs must _match_. If Host Build > Image Build, then Hyper-V isolation is the only option. And there's no base images for 20348 published yet to match the Windows Server 2022 Preview release on the Evaluation Center, so the latest Windows Server host we can use for process isolation is 20344 from the Windows Server Insiders Preview.

Reply:
I updated this PR to only add a new stage with the `DOCKER_WINDOWS_CONTAINERD_RUNTIME` env variable set, and we can use https://github.com/moby/moby/pull/42527 to update to more recent versions once things become available.